File: November 2022/29 November 2022/C++/Count the Number of Full Binary Trees.cpp
//{ Driver Code Starts
#include <bits/stdc++.h>
using namespace std;

// } Driver Code Ends
class Solution{
public:
    long long int numoffbt(long long int arr[], int n){
        long long int maxvalue = INT_MIN, minvalue = INT_MAX;
        for (int i = 0; i < n; i++){
            maxvalue = max(maxvalue, arr[i]);
            minvalue = min(minvalue, arr[i]);
        }
     
        long long int mark[maxvalue + 2]={0};
        long long int value[maxvalue + 2]={0};
        for (int i = 0; i < n; i++){
            mark[arr[i]] = value[arr[i]] = 1;
        }
        long long int ans = 0, mod=1000000007;
        for (long long int i = minvalue; i <= maxvalue; i++)
        {
            if (mark[i])
            {
                for (int j = i + i; j <= maxvalue && j/i <= i; j += i)
                {
                    if (!mark[j])
                        continue;
                    value[j] = (value[j] + (value[i] * value[j/i]))%mod;
                    if (i != j/i)
                        value[j] = (value[j] + (value[i] * value[j/i]))%mod;
                }
            }
            ans = (ans+value[i])%mod;
        }
        return ans;
    }
};


//{ Driver Code Starts.

int main() {
	// your code goes here
	int t;
	cin>>t;
	while(t--)
	{
		int n;
		cin>>n;
		long long int a[n];
		for(int i=0;i<n;i++)
		cin>>a[i];
		Solution ob;
		cout<<ob.numoffbt(a,n)<<endl;
		
	}
	return 0;
}


// } Driver Code Ends

File: November 2022/29 November 2022/Python/Count the Number of Full Binary Trees.py
#User function Template for python3

class Solution:
    def numoffbt(self, arr, n):
        maxvalue=-float("inf")
        minvalue=float("inf")
        for i in range(n):
            maxvalue=max(maxvalue,arr[i])
            minvalue=min(minvalue,arr[i])
        mark=[0]*(maxvalue+2)
        value=[0]*(maxvalue+2)
        for i in range(n):
            mark[arr[i]]=1
            value[arr[i]]=1
        ans=0
        mod=10**9+7
        for i in range(minvalue,maxvalue+1):
            if mark[i]:
                j=i+i
                while j<=maxvalue and (j//i)<=i:
                    
                    if not mark[j]:
                        j+=i
                        continue
                    value[j]=(value[j]+(value[i]*value[j//i]))%mod
                    if i!=j//i:
                        value[j]=(value[j]+(value[i]*value[j//i]))%mod
                    j+=i
            ans=(ans+value[i])%mod
        return ans


#{ 
 # Driver Code Starts
#Initial Template for Python 3

def main():

    T = int(input())

    while(T > 0):
        n = int(input())
        a = [int(x) for x in input().strip().split()]
        ob=Solution()
        print(ob.numoffbt(a, n))

        T -= 1


if __name__ == "__main__":
    main()





    
# } Driver Code Ends